### PR TITLE
Refactoring action-server 

### PR DESCRIPTION
**Pull request by med-hedi**
Signed-off-by: Mohamed Hedi Khefacha <mohamed.hedi@rte-france.com>

### CODE REVIEW

**Review by ChamseddineBhd — COMMENTED** [review]
same Name
 I could create two lists with the same name !

File: src/components/contingency-list.js [hunk 1]
```diff
@@ -216,42 +209,19 @@ const ContingencyLists = () => {
...
-            setCurrentFiltersContingency(null);
-            setCurrentItemType(scriptTypes.FILTERS);
-        }
-        setNewListName(name);
-        setNewListCreated(true);
-        setSelectedIndex(null);
-        setAlertEmptyList(false);
-        setOpenPopupNewList(false);
-        setBtnSaveListDisabled(false);
+    const handleEventPopupAddNewList = (event) => {
```
Comment (ChamseddineBhd) [inline src/components/contingency-list.js]: the parameter it's not an event, it's a boolean so rename it to openPopup or sthg else.

File: src/components/contingency-list.js [hunk 2]
```diff
@@ -310,19 +262,39 @@ const ContingencyLists = () => {
...
+                newListCreated ? name : currentItemName,
                 aceEditorContent
             );
         }
     };
 
-    const saveNewList = () => {
-        saveNewListResponse().then(() => {
+    const saveNewList = (name, type) => {
+        if (newListCreated) {
```
Comment (ChamseddineBhd) [inline src/components/contingency-list.js]: i think we can get rid of this state. we don't need it anymore;

**Review by ChamseddineBhd — CHANGES_REQUESTED** [review]

File: src/components/contingency-list.js [hunk 3]
```diff
@@ -179,37 +173,40 @@ const ContingencyLists = () => {
...
-    const [openPopupInfo, setOpenPopupInfo] = useState(false);
     const [openPopupConfirmDelete, setOpenPopupConfirmDelete] = useState(false);
 
-    const [equipmentID, setEquipmentID] = useState('*');
-    const [equipmentName, setEquipmentName] = useState('*');
-    const [equipmentType, setEquipmentType] = useState(equipmentTypes.LINE);
-    const [nominalVoltageOperator, setNominalVoltageOperator] = useState('=');
-    const [nominalVoltage, setNominalVoltage] = useState('');
-    const [countries, setCountries] = useState([]);
+    const emptyFiltersContingency = {
```
Comment (jonenst) [inline src/components/contingency-list.js]: move this outside of the render() function to avoid rerenders

Comment (ChamseddineBhd) [reply src/components/contingency-list.js:197]: it's already outside of render function

**Review by ChamseddineBhd — APPROVED** [review]

**Review by geofjamg — APPROVED** [review]

**Pull request merged**